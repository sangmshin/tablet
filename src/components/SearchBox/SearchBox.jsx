import React, { useState, memo, useEffect, useRef } from 'react';
import './SearchBox.module.scss';
import { Container } from 'react-bootstrap';
import PropTypes from 'prop-types';

const SearchBox = memo( ({ search }) => {
  const [ inputValue, setValue ] = useState('')
  const changeInput = e => setValue(e.target.value)
  const keyDown = e => inputValue && e.keyCode === 13 && search(inputValue)

  const textInput = useRef();
  useEffect(() => textInput.current.focus());
  
  return (
    <section>
      <Container>
        <form onSubmit={e => e.preventDefault()}>
          <legend>Where would you like to go?</legend>
          <input 
            value={inputValue} 
            onChange={changeInput} 
            onKeyDown={keyDown}
            ref={textInput}
            id="" 
            type="text" 
            name="location" 
          />
          <input 
            onClick={()=> inputValue && search(inputValue)} 
            // disabled={inputValue ? false : true}
            className="d-sm-none" 
            type="submit" 
            value="SEARCH"
          />
        </form>
      </Container>
    </section>
  )
})

SearchBox.propTypes = {
  search: PropTypes.func.isRequired,
};

SearchBox.defaultProps = {
  search: () => {},
};

export default SearchBox;
