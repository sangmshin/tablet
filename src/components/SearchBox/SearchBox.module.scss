@import 'styles/variables.scss';
@import 'styles/mixins.scss';


form {
  
  text-align: center;

  legend {
    font-size: 45px;
    @include bp($tablet){
      font-size: 30px;
    }
    @include bp($mobile){
      font-size: 10px;
    }

  }

  input {
    width: 100%;
    // background: transparent;
    border-top: transparent !important;
    border-left: transparent !important;
    border-right: transparent !important;
    border-bottom: black 1px solid !important;
    color: $orange;
    font-size: 40px;
    padding-left: 0;
    text-align: center;
    margin: auto 0;
  }

  input:focus {
    outline: none;
  }
}

// @media (max-width:600px){
//   legend{
//     font-size: 30px !important;
//   }
// }