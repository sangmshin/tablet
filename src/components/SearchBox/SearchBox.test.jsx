import React from 'react';
import SearchBox from './SearchBox';
import { shallow, render, mount } from 'enzyme';

it("renders correctly", () => {
  const wrapper = shallow(
    // eslint-disable-next-line react/react-in-jsx-scope
    <SearchBox search={()=>{}}/>
  )
  expect(wrapper).toMatchSnapshot()
})

it("renders correctly 2", () => {
  const wrapper = render(
    // eslint-disable-next-line react/react-in-jsx-scope
    <SearchBox search={()=>{}}/>
  )
  expect(wrapper).toMatchSnapshot()
})
