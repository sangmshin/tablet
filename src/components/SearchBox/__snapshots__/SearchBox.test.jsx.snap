// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <Container
    as="div"
    fluid={false}
  >
    <form
      onSubmit={[Function]}
    >
      <legend>
        Where would you like to go?
      </legend>
      <input
        id=""
        name="location"
        onChange={[Function]}
        onKeyDown={[Function]}
        type="text"
        value=""
      />
    </form>
  </Container>
</section>
`;

exports[`renders correctly 2 1`] = `
<section>
  <div
    class="container"
  >
    <form>
      <legend>
        Where would you like to go?
      </legend>
      <input
        id=""
        name="location"
        type="text"
        value=""
      />
    </form>
  </div>
</section>
`;
